### PR TITLE
Update for findAll, findRecord, peekAll, and peekRecord

### PR DESCRIPTION
**Pull request by HeroicEric**
Updated docs to match https://github.com/emberjs/rfcs/pull/27

Pending:

- [x] [Rename and deprecate finders (store.all -> store.peekAll and store.getById -> store.peekRecord)](https://github.com/emberjs/data/pull/3209)
- [x] [Make store.findAll(type) public](https://github.com/emberjs/data/pull/3234)
- [x] [Rename store.findByRecord to store.findRecord](https://github.com/emberjs/data/pull/3241)

### CODE REVIEW

**Comment by HeroicEric** [comment]
This should be good to go now

**Comment by trek** [comment]
@igorT can you review?

File: source/models/creating-and-deleting-records.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ var store = this.store;
 store.createRecord('post', {
   title: 'Rails is Omakase',
   body: 'Lorem ipsum',
-  author: store.find('user', 1)
+  author: store.findRecord('user', 1)
```
Comment (wecc) [inline source/models/creating-and-deleting-records.md]: Should this be `peekRecord`? Not sure how promises would be handled here...

File: source/models/finding-records.md [hunk 2]
```diff
@@ -1,44 +1,49 @@
...
+It's important to note that `DS.RecordArray` is not a JavaScript array.  It is
+an object that implements [`Ember.Enumerable`][1]. This is important because,
+for example, if you want to retrieve records by index, the `[]` notation will
+not work--you'll have to use `objectAt(index)` instead.
+
+[1]: http://emberjs.com/api/classes/Ember.Enumerable.html
+
+### Querying for Multiple Records
 
 If you provide a plain object as the second argument to `find`, Ember Data will
```
Comment (wecc) [inline source/models/finding-records.md]: `query` instead of passing POJO to `find`

**Comment by wecc** [comment]
LGTM

**Comment by michaelrkn** [comment]
@HeroicEric can you rebase please?

**Comment by HeroicEric** [comment]
@michaelrkn I can rebase later tonight

**Comment by locks** [comment]
@rwjblue merge this puppy please

**Comment by michaelrkn** [comment]
@rwjblue any reason not to merge this in?

**Pull request merged**